### PR TITLE
Remove TransactionConfig

### PR DESCRIPTION
**Pull request by nambrot**
This PR removes TransactionConfig which has become equivalent to IChainConnection.

This PR also makes `provider` required as a chain connection is kind of useless without one.

### CODE REVIEW

File: typescript/sdk/src/chains.ts [hunk 1]
```diff
@@ -4,6 +4,35 @@ import { ethers } from 'ethers';
...
+  provider: new StaticCeloJsonRpcProvider('https://forno.celo.org', 42220),
+  confirmations: 1,
+};
+
+export const polygon: IChainConnection = {
+  provider: new ethers.providers.JsonRpcProvider(
+    'https://rpc-mainnet.matic.quiknode.pro',
+    137,
+  ),
+  confirmations: 200,
```
Comment (tkporter) [inline typescript/sdk/src/chains.ts:25]: fwiw I've seen deeper reorgs than this, but this is for just non-core use right? in other words, does this # ever get passed in as config to the agent deployments done by `deploy`?

File: typescript/sdk/src/provider.ts [hunk 2]
```diff
@@ -31,17 +31,13 @@ export class ChainConnection {
...
-  ) {
-    const params = Array.isArray(chainConnectionConfigs)
-      ? chainConnectionConfigs.map((v) => [v, {}])
-      : (Object.entries(chainConnectionConfigs) as [Chain, IChainConnection][]);
-    const providerEntries = params.map(([chain, v]) => [
-      chain,
-      new ChainConnection(v),
-    ]);
-    super(Object.fromEntries(providerEntries));
+  constructor(chainConnectionConfigs: ChainMap<Chain, IChainConnection>) {
```
Comment (tkporter) [inline typescript/sdk/src/provider.ts:34]: this is so much nicer to read

**Review by tkporter — APPROVED** [review]

File: typescript/sdk/src/chains.ts (hunk 1 above)
Comment (nambrot) [reply typescript/sdk/src/chains.ts:25]: It does look like we now source the number of confirmations from `AgentConfig`, though I would have intuitively thought we source them from the same place. IMO we should, wdyt @mattiecnvr @webbhorn ? 

I thought 200 was the max reorg depth for polygon, what do you think is the right safe period?

Comment (nambrot) [reply typescript/sdk/src/chains.ts:25]: https://twitter.com/koeppelmann/status/1452603141971759106?lang=en

Comment (mattiecnvr) [reply typescript/sdk/src/chains.ts:25]: Yes, we should be consistent on where the information is defined. Easier to keep track of.

**Pull request merged**